pkg/user/user_integration_test.go

```diff
@@ -152,99 +237,304 @@ func TestUserHandler(t *testing.T) {
+
+				// TODO: Assert response body?
+			}
+
+			{
+				t.Log("RefreshTokensUsingCookieWithRememberMe")
+
+				_, email, password := createUser(t, client, userService)
+				var tokens *token.Tokens
+				client.PostJSON(t, "/tokens", strings.NewReader(`{}`), &tokens, inttest.WithBasicAuth(email, password))
+				require.NotEmpty(t, tokens.AccessToken, "should return an access token")
+				request := client.NewRequest(t, http.MethodPost, "/refresh", strings.NewReader(`{}`), inttest.WithHeader("Content-Type", "application/json"))
+				refreshCookie := &http.Cookie{Name: "refreshToken", Value: tokens.RefreshToken, Path: "/refresh"}
+				require.NoError(t, refreshCookie.Valid())
+				request.AddCookie(refreshCookie)
+				rememberMeCookie := &http.Cookie{Name: "rememberMe", Value: "true", Path: "/refresh"}
+				require.NoError(t, rememberMeCookie.Valid())
+				request.AddCookie(rememberMeCookie)
+
+				response, err := client.Client.Do(request)
+				require.NoError(t, err)
 
-				client.Do(t, http.MethodPost, "/tokens", nil, http.StatusUnauthorized, inttest.WithBasicAuth("user3@dhis2.org", "oneoneoneoneoneoneone111"))
+				assert.Equal(t, http.StatusCreated, response.StatusCode)
+				actualCookies := response.Cookies()
+				require.Len(t, actualCookies, 3)
+				accessTokenCookie := findCookieByName("accessToken", actualCookies)
+				require.NotNil(t, accessTokenCookie)
+				assert.Equal(t, accessTokenCookie.Path, "/")
+				assert.Equal(t, accessTokenCookie.MaxAge, 10)
+				assert.Equal(t, accessTokenCookie.Secure, true)
+				assert.Equal(t, accessTokenCookie.HttpOnly, true)
+				assert.Equal(t, accessTokenCookie.SameSite, http.SameSiteStrictMode)
+				refreshTokenCookie := findCookieByName("refreshToken", actualCookies)
+				require.NotNil(t, refreshTokenCookie)
+				assert.Equal(t, refreshTokenCookie.Path, "/refresh")
+				assert.Equal(t, refreshTokenCookie.MaxAge, 30)
+				assert.Equal(t, refreshTokenCookie.Secure, true)
+				assert.Equal(t, refreshTokenCookie.HttpOnly, true)
+				assert.Equal(t, refreshTokenCookie.SameSite, http.SameSiteStrictMode)
+				rememberMeTokenCookie := findCookieByName("rememberMe", actualCookies)
+				require.NotNil(t, rememberMeTokenCookie)
+				assert.Equal(t, rememberMeTokenCookie.Path, "/refresh")
+				assert.Equal(t, rememberMeTokenCookie.MaxAge, 30)
+				assert.Equal(t, rememberMeTokenCookie.Secure, true)
+				assert.Equal(t, rememberMeTokenCookie.HttpOnly, true)
+				assert.Equal(t, rememberMeTokenCookie.SameSite, http.SameSiteStrictMode)
+
+				// TODO: Assert response body?
 			}
+
+			{
+				t.Log("RefreshTokensRequestBody")
+
+				_, email, password := createUser(t, client, userService)
+				var tokens *token.Tokens
+				client.PostJSON(t, "/tokens", strings.NewReader(`{}`), &tokens, inttest.WithBasicAuth(email, password))
+				require.NotEmpty(t, tokens.AccessToken, "should return an access token")
+				requestBody := strings.NewReader(fmt.Sprintf(`{"refreshToken": "%s"}`, tokens.RefreshToken))
```

Comment:
since we are doing something like that often a helper like

```
func jsonBody(format string, args ...any) io.Reader {
  return strings.NewReader(fmt.Sprintf(format, args...))
}
```

(or name it just `body` as its agnostic to what you put in)

makes sense to have it read like

```
request := client.NewRequest(t, http.MethodPost, "/refresh", jsonBody(`{"refreshToken": "%s"}`, tokens.RefreshToken), inttest.WithHeader("Content-Type", "application/json"))
```